manual/src/cmds/runtime-tracing.etex

```diff
@@ -0,0 +1,316 @@
+
+We could remove "Runtime_events.start ();" from the previous example and,
+instead, call the program as below to produce the same result:
+
+\begin{verbatim}
+OCAML_RUNTIME_EVENTS_START=1 ./example
+\end{verbatim}
+
+\subsubsection{s-runtime-tracing-environment-variables}{Environment variables}
+
+Environment variables can be used to control different aspects of the runtime
+event tracing system. The following environment variables are available:
+
+\begin{itemize}
+  \item OCAML_RUNTIME_EVENTS_START if set will cause the runtime events system
+  to be started as part of the OCaml runtime initialization.
+  \item OCAML_RUNTIME_EVENTS_DIR sets the directory where the ".events"
+  files containing the runtime event tracing system's ring buffers will be located.
+  If not present the program's working directory will be used.
+  \item OCAML_RUNTIME_EVENTS_PRESERVE if set will make the OCaml runtime
+  preserve the runtime events ring buffer files past the termination of the OCaml program.
+  This can be useful for monitoring very short running programs.
+  If not set, the ".events" files of the OCaml program will be deleted
+  at program termination.
+\end{itemize}
+
+The size of the runtime events ring buffers can be configured via OCAMLRUNPARAM,
+see section \ref{s:ocamlrun-options} for more information.
+
+\subsubsection{s-runtime-tracing-instrumented-runtime}{Building with the instrumented runtime}
+
+To receive events that are only available in the instrumented runtime, the
+OCaml program needs to be compiled and linked against the instrumented runtime.
+For our example program from earlier, this is achieved as follows:
+
+\begin{verbatim}
+ocamlopt -runtime-variant i -I +runtime_events -I +unix unix.cmxa runtime_events.cmxa example.ml -o example
+\end{verbatim}
+
+And for dune:
+
+\begin{verbatim}
+(executable
+ (name example)
+ (modules example)
+ (flags "-runtime-variant=i")
+ (libraries unix runtime_events))
+\end{verbatim}
+
+\subsection{s-runtime-tracing-tooling}{With tooling}
+
+Programmatic access to events is intended primarily for writers of observability
+libraries and tooling that end-users use. The flexible API enables use of
+the performance data from runtime events for logging and monitoring purposes.
+
+In this section we cover several utilities in the \texttt{runtime_events_tools}
+package which provide simple ways of extracting and summarisingdata from runtime
+events. The trace utility in particular produces similar data to the previous
+'eventlog' instrumentation system available in OCaml 4.12 to 4.14.
+
+First, install \texttt{runtime_events_tools} in an OCaml 5.0+ opam switch:
+
+\begin{verbatim}
+opam install runtime_events_tools
+\end{verbatim}
+
+This should install the olly tool in your path. You can now generate
+runtime traces for programs compiled with OCaml 5.0+ using the trace subcommand:
+
+\begin{verbatim}
+olly trace trace.json your_program.exe .. args ..
```

Comment:
Command-line arguments to `olly` need to be quoted to prevent them from being interpreted as arguments to `olly`:
```
  olly trace trace.json 'your_program.exe .. args ..'
```